plot/plot.app.js

```diff
@@ -684,13 +690,42 @@
 
                                                     data.forEach(function (row) {
                                                         if(label) {
-                                                            values.labels.push(row[trace.legend_col]);
+                                                            var traceLabel=row[trace.legend_col];
+
+                                                            if(trace.hasOwnProperty("legend_markdown_pattern") && trace.legend_markdown_pattern){
+                                                                var qCharacter = /\((.*?)\)/ig.exec(trace.legend_markdown_pattern)[1] !== -1 ? "&" : "?";
+                                                                $rootScope.templateParams= {
+                                                                    Schema_Table: row['Schema_Table'],
+                                                                    Data_Type_Filter: ERMrest.encodeFacet(row['Data_Type_Filter']),
+                                                                    Title: traceLabel,
+                                                                    Context_Params: qCharacter+"pcid="+contextUrlParams.cid+"&ppid="+contextUrlParams.pid
+                                                                };
+                                                                traceLabel=ERMrest.renderMarkdown(ERMrest.renderHandlebarsTemplate(trace.legend_markdown_pattern,$rootScope.templateParams),true);
+                                                                var matchs =/<a\b.+href="([^\n\r]*)"\s/ig.exec(traceLabel);
+                                                                values.legend_markdown_pattern.push(matchs[1]);
+                                                                delete $rootScope.templateParams;
+                                                            }
+                                                            values.labels.push(traceLabel);
+                                                            values.text.push(row[trace.legend_col]);
+                                                            console.log(values.text)
                                                         }
                                                         values.values.push(formatData(row[trace.data_col], plot.config ? plot.config.format_data : false, "pie"));
+                                                    
                                                     });
                                                     plot_values.data.push(values);
+                                                    if(plot.config.hasOwnProperty("title_hyperlink_pattern")){
```

Comment:
Rendering the markdown template pattern for Main title,xaxis title and y axis title if available